managedtenants/core/addons_loader/addon.py

```diff
@@ -140,6 +142,29 @@ def _validate_additional_catalogue_srcs(self, metadata):
                     " should have a unique name"
                 )
 
+    def _validate_secret_names(self, metadata):
```

Comment:
```suggestion
    def _validate_secret_names(self):
```
and inside any under the function let's use `self.metadata` as this is a private method meant to fiddle with the object's attributes.

If it's operating on NONE of the objects attributes then, defining it as a static method or a helper function would've made more sense :)

Reply:
Few findings regarding this:

- these validations are done under the `load_metadata` function here: https://github.com/mt-sre/managed-tenants-cli/blob/6784be01387e22079870092d68a7de02d149ddc0/managedtenants/core/addons_loader/addon.py#L113
- and this `load_metadata` function is used to assign the `metadata` attribute of the `Addon` object: https://github.com/mt-sre/managed-tenants-cli/blob/6784be01387e22079870092d68a7de02d149ddc0/managedtenants/core/addons_loader/addon.py#L36
- so if we access `self.metadata` inside this function call, then it raises this error: `'Addon' object has no attribute 'metadata'`

Therefore, we can keep it the same way :)